workers/unity/Assets/Gdk/Core/GameObjectRepresentation/EntityGameObjectLifecycleSystem.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    public class EntityGameObjectLifecycleSystem : ComponentSystem
+    {
+        private uint currentHandle;
+        private readonly Dictionary<uint, GameObject> entityGameObjectCache = new Dictionary<uint, GameObject>();
+
+        public struct AddedEntitiesData
+        {
+            public int Length;
+            public EntityArray Entities;
+            [ReadOnly] public ComponentArray<GameObjectReference> GameObjectReferences;
+            public SubtractiveComponent<GameObjectReferenceHandle> NoGameObjectReferenceHandle;
+        }
+
+        public struct RemovedEntitiesData
+        {
+            public int Length;
+            public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<GameObjectReferenceHandle> GameObjectReferenceHandles;
+            public SubtractiveComponent<GameObjectReference> NoGameObjectReference;
+        }
+
+        [Inject] private AddedEntitiesData addedEntitiesData;
+        [Inject] private RemovedEntitiesData removedEntitiesData;
+
+        protected override void OnUpdate()
+        {
+            for (int i = 0; i < addedEntitiesData.Length; i++)
+            {
+                var handle = currentHandle++;
+                var gameObjectReferenceHandleComponent = new GameObjectReferenceHandle { GameObjectHandle = handle };
+                PostUpdateCommands.AddComponent(addedEntitiesData.Entities[i], gameObjectReferenceHandleComponent);
```

Comment:
Mild style preference - I prefer the PostUpdateCommands to be at the end of the block. Makes it clear what entity changes will happen as a result of this.

Reply:
Changed order.